### DISCUSSION TITLE
How to enable Python capability of iterating over an object while TF eager execution is disabled?

### DISCUSSION THREAD

**Discussion by Saif-KF** [Trusted-AI/adversarial-robustness-toolbox#1935] [Q&A]
How can I iterate over an object while I the eager execution is disabled?
Utilizing the code in the Notebook named "[ART for TensorFlow v2 - Keras API](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/notebooks/art-for-tensorflow-v2-keras.ipynb)", I could not extract the the training subset of a custom dataset (saved in "train_ds" object) to images and labels. I got an error "RuntimeError: `tf.data.Dataset` only supports Python-style iteration in eager mode or within tf.function.". Could you help on how to address this?

**Comment by beat-buesser** [comment]
Hi @Saif-KF I think this sounds like a questions for the TensorFlow team.
